pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -83,93 +76,93 @@ func (c *ClusterMemberController) sync(ctx context.Context, syncCtx factory.Sync
 	return updateErr
 }
 
+func (c *ClusterMemberController) ensureEtcdLearnerPromotion(ctx context.Context, recorder events.Recorder) error {
+	members, err := c.etcdClient.MemberList(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get etcd member list: %v", err)
+	}
+
+	var errs []error
+	for _, member := range members {
+		// Promote learner to voting member if log is in sync with leader.
+		if member.IsLearner {
+			err := c.etcdClient.MemberPromote(ctx, member.ID)
+			// TODO: this is not working as expected
+			if err != nil && errors.Is(err, etcdserver.ErrLearnerNotReady) {
+				recorder.Warningf("ScaleUpCluster", "promotion failed etcd learner is not yet ready %s", member.PeerURLs[0])
+				continue
+			}
+			if err != nil {
+				//errs = append(errs, err)
```

Comment:
I think this makes sense as we return the aggregated errors below.

Reply:
agreed I just have an error handling issue above here that needs to get worked out.